src/configure/configurers/ResourceSelectorFactory.ts

```diff
@@ -0,0 +1,20 @@
+import { PipelineConfiguration, TargetResourceType } from "../model/models";
+import { AksAzureResourceSelector } from "./AksAzureResourceSelector";
+import { IAzureResourceSelector } from "./IAzureResourceSelector";
+import { WebAppAzureResourceSelector } from "./WebAppAzureResourceSelector";
+
+export class ResourceSelectorFactory {
+
+    public static getAzureResourceSelector(pipelineConfiguration: PipelineConfiguration): IAzureResourceSelector {
+        if (pipelineConfiguration.template.label === "Containerized application to AKS") {
```

Comment:
If I am not wrong, the template is not yet decided, then why depend on `pipelineConfiguration.template`? We should rather use `this.inputs.potentialTemplates`. No?

Reply:
The label is decided. Potential templates will have all the labels having the same label.